src/core/topology.rs

```diff
@@ -20,17 +20,56 @@
 
 //! Mixnet topology interface.
 
-use crate::{core::MixPublicKey, MixPeerId};
-
-/// Provide network topology information to the mixnet.
-pub trait Topology: Send + 'static {
-	/// Select a random recipient for the message to be delivered. This is
-	/// called when the user sends the message with no recipient specified.
-	/// E.g. this can select a random validator that can accept the blockchain
-	/// transaction into the block.
-	/// Return `None` if no such selection is possible.
-	fn random_recipient(&self) -> Option<MixPeerId>;
-
-	/// For a given peer return a list of peers it is supposed to be connected to.
-	fn neighbors(&self, id: &MixPeerId) -> Vec<(MixPeerId, MixPublicKey)>;
+use crate::{core::MixPublicKey, MixPeerId, core::MixPeerAddress};
+use rand::{prelude::{SliceRandom, IteratorRandom}, CryptoRng, Rng};
+
+/// Contains network information. Current implementation assumes that each node
+/// in the set is connected to every other node.
+#[derive(Default)]
+pub struct SessionTopology {
+	nodes: Vec<(MixPeerId, MixPublicKey, Vec<MixPeerAddress>)>,
+}
+
+impl SessionTopology {
+	pub fn new(nodes: Vec<(MixPeerId, MixPublicKey, Vec<MixPeerAddress>)>) -> Self {
+		Self { nodes }
+	}
+
+	pub fn random_recipient<R: Rng + CryptoRng + ?Sized>(&self, rng: &mut R) -> Option<MixPeerId> {
+		self.nodes.iter().choose(rng).map(|(id, _, _)| id.clone())
+	}
+
+	pub fn gateways<R: Rng + CryptoRng + ?Sized>(&self, rng: &mut R) -> Vec<MixPeerAddress> {
```

Comment:
IIUC this is to be able to connect to node when being not part of the session validator set,
and afterward we just would need to do random path, with a forced first node depending on our current open connection.

Reply:
Right, I'll add documentation